src/ui/map.js

```diff
@@ -2606,16 +2608,23 @@ class Map extends Camera {
         return this.style.getFeatureState(feature);
     }
 
-    _containerDimensions() {
-        let width = 0;
-        let height = 0;
-
+    _containerDimensions(): any {
         if (this._container) {
-            width = this._container.getBoundingClientRect().width || 400;
-            height = this._container.getBoundingClientRect().height || 300;
-        }
+            const width = this._container.getBoundingClientRect().width;
+            const height = this._container.getBoundingClientRect().height;
+
+            // If css transform(scale) property value was applied, getBoundingClientRect will return its values after calculating with transform.
+            // In order to circumvent this scenario, the difference between getBoundingClientRect and clientWidth/clientHeight is used to get the expected width/height values.
+            if (Math.round(width) !== this._container.clientWidth || Math.round(height) !== this._container.clientHeight) {
+                this._containerWidth = width / (width / this._container.clientWidth) || 400;
```

Comment:
I think this can be simplified
```suggestion
                this._containerWidth = this._container.clientWidth || 400;
```

Reply:
Thanks for the review @SnailBones  - I'll definitely add in a unit test.

So clientWidth is different from width (which is getBoundingClientRect().width)... this is the case when getBoundingClientRect().width is affected by the transform property so in order to undo the scale, the difference between clientWidth (which isn't affected by transform) and getBoundingClientRect().width is then applied to getBoundingClientRect().width to undo the changes from transform.

But I think it's a definitely valid option to just completely replace getBoundingClientRect.width/height with clientWidth/Height in the case transform is used, but these cases wouldn't be as precise, but doing the calculation also affects the precision as well. What do you think @arindam1993 ?

Reply:
> But I think it's a definitely valid option to just completely replace getBoundingClientRect.width/height with clientWidth/Height in the case transform is used,

I think this is what you're already doing?
x / ( x / y ) = y

Reply:
:woman_facepalming: